print-bgp.c

```diff
@@ -2496,6 +2500,63 @@ bgp_attr_print(netdissect_options *ndo,
             len -= 12;
         }
         break;
+    case BGPTYPE_BGPSEC_PATH:
+    {
+        uint16_t sblen, splen, siglen;
+        u_int j, idx;
+
+        splen = GET_BE_U_2(tptr);
+        idx = 0;
+
+        if (splen < 8) {
+            ND_PRINT(" [secure path too short]");
+            tptr += splen;
+            break;
+        }
+
+        ND_PRINT("\n\t    Secure Path Length: %u", splen);
+
+        idx += 2;
+        splen -= 2;
+        /* parse secure path segments */
+        while (splen > 0) {
+            ND_PRINT("\n\t      Secure Path Segment: pCount: %u, Flags: %u, AS: %u",
+                     GET_U_1(tptr + idx),
+                     GET_U_1(tptr + idx + 1),
+                     GET_BE_U_4(tptr + idx + 2));
+            idx += 6;
+            splen -= 6;
+        }
+
+        sblen = GET_BE_U_2(tptr + idx);
+        ND_PRINT("\n\t    Signature Block: Length: %u, Algo ID: %u",
+                 sblen,
+                 GET_U_1(tptr + idx + 2));
+        idx += 3;
+        sblen -= 3;
+
+        /* parse signature segments */
+        while (sblen > 0) {
+            siglen = GET_BE_U_2(tptr + idx + 20);
+            ND_PRINT("\n\t      Signature Segment:\n\t        SKI: ");
+            for (j = 0; j < 20; j++) {
+                ND_PRINT("%02x ", GET_U_1(tptr + idx + j));
```

Comment:
This loop seems highly inefficient and ugly, but I couldn't get `hex_print` to work... I'd be happy about some guidance.

Reply:
Something like:
```patch
hex_print(ndo, "\n\t          ", tptr + idx, 20);
```
?

Reply:
Thanks, looks like this works like a charm.